engine/src/player-platform.cpp

```diff
@@ -1831,8 +1831,8 @@ void MCPlayer::setenabledtracks(uindex_t p_count, uint32_t *p_tracks_id)
 			
             for (uindex_t i = 0; i < t_track_count; i++)
             {
-				// If the list of enabledtracks we set contains empty/0, just skip it
-				if (p_tracks_id[i] == 0)
+				// If the list of enabledtracks we set contains 0 (empty), just skip it until it is non-empty
+				while (p_tracks_id[i] == 0)
 					i++;
```

Comment:
You will need to have some sort of guard here so that `i` doesn't exceed `t_track_count` (in fact that should have been there before also)

In fact the easiest way to fix this would be to change the `i++;` to `continue;` instead of changing from the `if` to a `while` loop